docs/tools/hardhat/hardhat-zksync-vyper-verify.md

```diff
@@ -0,0 +1,154 @@
+When executed in this manner, the verification task attempts to compare the compiled bytecode of all the contracts in your local environment with the deployed bytecode of the contract you are seeking to verify. If there is no match, it reports an error.
+
+```sh
+yarn hardhat vyper:verify --network <network> <contract address> --contract <fully qualified name>
+```
+
+With the `--contract` parameter you can also specify which contract from your local setup you want to verify by specifying its Fully qualified name. Fully qualified name structure looks like this: "contracts/Contract.vy:Contract" <br/>
+
+#### Constructor arguments
+
+If your contract was deployed with the specific constructor arguments, you need to specify them when running the verify task. For example:
+
+```sh
+yarn hardhat vyper:verify --network testnet 0x7cf08341524AAF292255F3ecD435f8EE1a910AbF "Hi there!"
+```
+
+If your constructor takes a complex argument list, you can write a separate javascript module to export it. For example, create an `arguments.js` file with the following structure:
+
+```typescript
+module.exports = [
+  "a string argument",
+  "0xabcdef",
+  "42",
+  {
+    property1: "one",
+    property2: 2,
+  },
+];
+```
+
+Include it in the verify function call by adding a new parameter: `--constructor-args arguments.js`:
+
+```sh
+yarn hardhat vyper:verify --network testnet 0x7cf08341524AAF292288F3ecD435f8EE1a910AbF --constructor-args arguments.js
+```
+
+The hardhat-zksync-verify plugin also supports the verification with encoded constructor parameters.
+
+In order to use the encoded parameters, you need to specify a separate javascript module and export them as a <b>_non-array_</b> parameter.
+It is important for encoded arguments to start with `0x` in order to be recognized by the plugin. For example:
+
+```typescript
+module.exports = "0x0x00087a676164696a61310000087a676164696a61310000000000000000000000008537b364a83f5c9a7ead381d3baf9cbb83769bf5";
+```
+
+### Verification status check
+
+The verification process consists of two steps:
+
+- A verification request is sent to confirm if the given parameters for your contract are correct.
+- Then, we check the verification status of that request.
+  Both steps run when you run the `vyper:verify` task, but you will be able to see your specific verification request ID.
+  You can then use this ID to check the status of your verification request without running the whole process from the beginning.
+
+The following command checks the status of the verification request for the specific verification ID:
+
+```sh
+yarn hardhat verify-status:vyper --verification-id <your verification id>
+```
+
+### Verify smart contract programmatically
+
+If you need to run the verification task directly from your code, you can use the hardhat `verify:verify:vyper` task with the previously mentioned parameters with the difference in using `--address` parameter when specifying contarct's address.
```

Comment:
```suggestion
If you need to run the verification task directly from your code, you can use the hardhat `verify:verify:vyper` task with the previously mentioned parameters with the difference in using `--address` parameter when specifying contract's address.
```

Typo although I don't understand what this part means: " with the difference in using `--address` parameter when specifying contract's address."